docs/zkapps/tutorials/04-in-browser-react-ui.mdx

```diff
@@ -0,0 +1,417 @@
+
+This example uses an RPC endpoint. An in-browser Mina node is in the works, which will provide full node level security to your users, in the browser.
+
+:::caution
+This tutorial currently uses localStorage to store and manage the user's private key. While helpful for prototyping applications, we do not recommend using it in production, since it does not provide only limited guarantees around persistence and security. 
+
+A new version of SnarkyJS is in the works which will connect to Auro wallet for key management in production applications.
+:::
+
+## Project setup
+
+First, setup a new project with
+
+```sh
+zk project 04-zkapp-browser-ui --ui next
+```
+
+On the prompts that appear, select:
+1. Do you want your NextJS project to use TypeScript? `yes`
+2. Do you want to setup your project for github pages? `yes`
+
+This will create a new project with two folders instead of just one:
+* `contract`: This stores your smart contract code
+* `ui`: This is where you will write your ui
+
+We will be using the default contract (`Add`), so all we'll be doing is building it for use from our ui.
+
+To do this, enter the contract folder, and run 
+```sh
+npm run build
+```
+
+In making your own zkApp, you will need to edit the contract folder, and rebuild before accessing it from your UI.
+
+## Implementing the UI
+
+Our React UI will have a few components. First, there is the react page itself. In addition though, there is code that uses SnarkyJS.
+
+Because SnarkyJS code is computationally intensive, running it as usual in a script would block our browser's UI thread, causing our page to become unresponsive at times. To solve this, we will put our SnarkyJS code in a webworkers. 
+
+This will be implemented via 2 helper files:
+* `zkappWorker.ts`: The web worker code itself
+* `zkappWorkerClient.ts`: Code that we run from react to interact with the web worker.
+
+We encourage you to read this files, to understand how they work and to extend them for your own zkApp, but we will not review them in detail here, in favor of spending time on the react code.
+
+Download these files from [here](https://raw.githubusercontent.com/es92/zkApp-examples/main/04-zkapp-browser-ui/ui/pages/zkappWorker.ts) and [here](https://raw.githubusercontent.com/es92/zkApp-examples/main/04-zkapp-browser-ui/ui/pages/zkappWorkerClient.ts), and place them in your `ui/pages` folder.
+
+### globals.css
+
+We will be using an `<a>` link in our application - add the following to the end of `styles/globals.css` to style the link:
+```css
+a {
+  color: blue;
+}
+```
+
+### Running the react app
+
+To run the react app, open up two new terminals. In one, type:
+```sh
+npm run dev
+```
+
+And in the other, 
+```sh
+npm run ts-watch
```

Comment:
When was this installed? 

I think we should add this to the project template in the zkapp-cli. I'll create an issue